x-pack/plugins/infra/public/hooks/use_trial_state.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { API_BASE_PATH as LICENSE_MANAGEMENT_API_BASE_PATH } from '../../../license_management/common/constants';
+
+export enum TrialStatusLoadState {
+  Loading = 'loading',
+  Ok = 'ok',
+  Error = 'error',
+}
+
+interface UseTrialStatusState {
+  loadState: TrialStatusLoadState;
+  isTrialAvailable: boolean;
+}
+
+export function useTrialStatus(): UseTrialStatusState {
+  const { services } = useKibana();
+
+  const [loadState, setLoadState] = useState<TrialStatusLoadState>(TrialStatusLoadState.Loading);
+  const [isTrialAvailable, setIsTrialAvailable] = useState<boolean>(false);
+
+  useEffect(() => {
+    async function fetchTrial() {
+      try {
+        const response = await services.http.get(`${LICENSE_MANAGEMENT_API_BASE_PATH}/start_trial`);
```

Comment:
As this hits an external API, could we add some runtime type protection with io-ts and our `decodeOrThrow` helper. E.g.: https://github.com/elastic/kibana/blob/master/x-pack/plugins/infra/public/containers/logs/log_analysis/api/validate_datasets.ts#L35 , so that we're explcit about the data we expect back, and we fail with a more relevant error (in cases where it's type mismatch related).